### PR TITLE
Remove deprecated calls in cumulus-parachain-system

### PR DESCRIPTION
**Pull request by gui1117**
Calls were written to be removed after June 2024. This PR removes them.

This PR will break users using those calls. The call won't be decodable by the runtime, so it should fail early with no consequences. The functionality must be same as before, users will just need to use the calls in `System`.

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]
🙏

File: cumulus/pallets/parachain-system/src/lib.rs [hunk 1]
```diff
@@ -951,30 +904,6 @@ pub mod pallet {
...
-					return Ok(ValidTransaction {
-						priority: 100,
-						requires: Vec::new(),
-						provides: vec![hash.as_ref().to_vec()],
-						longevity: TransactionLongevity::max_value(),
-						propagate: true,
-					})
-				}
-			}
-			if let Call::set_validation_data { .. } = call {
```
Comment (muharem) [inline cumulus/pallets/parachain-system/src/lib.rs:972]: We need this (=

Comment (gui1117) [reply cumulus/pallets/parachain-system/src/lib.rs:972]: `set_validation_data` is an inherent, inherent should not be validated by `validate_unsigned`, they only go through `pre_dispatch`.

The default implementation (by construct_runtime) is to consider all calls invalid in `validate_unsigned` and all calls valid in `pre_dispatch`. So the default implementation is what we want here.

Comment (georgepisaltu) [reply cumulus/pallets/parachain-system/src/lib.rs:972]: Indeed the behavior in construct runtime is [allow calls in `pre_dispatch`](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/frame/support/procedural/src/construct_runtime/expand/unsigned.rs#L69) and [automatically reject any calls that don't have `ValidateUnsigned::validate_unsigned` impl](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/frame/support/procedural/src/construct_runtime/expand/unsigned.rs#L84).

On a different note, inherents are never "validated" as transactions and never enter the transaction pool, they are just included in the block and [applied along with other regular transactions](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/frame/executive/src/lib.rs#L654). Specifically when [inherents are applied](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/frame/executive/src/lib.rs#L801), the [`Applyable` impl of `UncheckedExtrinsic`](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/primitives/runtime/src/generic/checked_extrinsic.rs#L71) calls on [`ValidateUnsigned::pre_dispatch`](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/primitives/runtime/src/generic/checked_extrinsic.rs#L81). `ValidateUnsigned::validate_unsigned` is called only when [validating a checked extrinsic](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/substrate/frame/executive/src/lib.rs#L883) in `frame_executive`, which is called when a transaction tries to enter the transaction pool, which should never happen for inherents.

`set_validation_data` [must be called exactly once per block, as an inherent, at the beginning of the block](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/cumulus/pallets/parachain-system/src/lib.rs#L508). This means it was never correct to have a `ValidateUnsigned` validation for it. It should never be allowed as an unsigned transaction, only as an inherent.

Also, for peace of mind, all of the tests, including zombienet tests, are passing with the changes in this PR. [The chain doesn't work without calls to `set_validation_data` every block](https://github.com/paritytech/polkadot-sdk/blob/9fecd89880bebec86815e3413327dd8a5f80befd/cumulus/pallets/parachain-system/src/lib.rs#L266), so the fact that the tests pass reinforces the reasoning.

**Review by georgepisaltu — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
Thanks for contributing once again @gui1117 🙏

**Comment by gui1117** [comment]
CI is good, non-mandatory `gitlab-subsystem-benchmark-approval-voting ` is failing, but it must not be related to the PR.

**Pull request merged**